topostats/run_topostats.py

```diff
@@ -163,12 +164,12 @@ def main(args=None):
 
     # Write sample configuration if asked to do so and exit
     if args.create_config_file:
-        write_yaml(
-            config,
-            output_dir="./",
-            config_file=args.create_config_file,
-            header_message="Sample configuration file auto-generated",
-        )
+        default_config_path = pkg_resources.path(__package__, "default_config.yaml")
```

Comment:
I understand why this is here but I think we can make things a little simpler by changing some code earlier on.

Currently on lines 146-150 we have...

```python
    if args.config_file is not None:
        config = read_yaml(args.config_file)
    else:
        default_config = pkg_resources.open_text(__package__, "default_config.yaml")
        config = yaml.safe_load(default_config.read())
```

...and so we have two lines that are very similar here as they both have `pkg_resources.open_text(__package__, "default_config.yaml")`.

If we change the logic of the earlier code we only need to have one line for setting the path to the `default_config.yaml`...

```python
    if args.config_file is None:
        default_config = pkg_resources.open_text(__package__, "default_config.yaml")
        config = yaml.safe_load(default_config.read())
    else:
        config = read_yaml(args.config_file)
```

...and the above line is no longer needed (although line 172 would need changing to use `default_config` rather than `default_config_path`).

Reply:
I could be misunderstanding you. 
```python
pkg_resources.open_text(__package__, "default_config.yaml")
```
returns a string of the contents of the config file, which is not usable by `shutil.copy`, which takes a file system path. This could work if I manually saved to the file using `with open() as f`, is this what you intended?